packages/core/src/internal/validation/futures/validateNamedContractDeployment.ts

```diff
@@ -80,6 +81,7 @@ export async function validateNamedContractDeployment(
   if (isModuleParameterRuntimeValue(future.value)) {
     const param =
       deploymentParameters[future.value.moduleId]?.[future.value.name] ??
+      deploymentParameters.$global[future.value.name] ??
```

Comment:
```suggestion
      deploymentParameters.$global?.[future.value.name] ??
```

I think we need to guard against there being no global parameters passed.